src/NUnitFramework/framework/Internal/TestNameGenerator.cs

```diff
@@ -231,7 +231,30 @@ protected static string GetDisplayString(object arg, int stringMax)
                     ? "null"
                     : Convert.ToString(arg, System.Globalization.CultureInfo.InvariantCulture);
 
-                if (arg is double)
+                Array argArray = arg as Array;
+                if (argArray != null)
+                {
+                    if (argArray.Length == 0)
+                        display = "[]";
+                    else
+                    {
+                        StringBuilder builder = new StringBuilder();
+                        builder.Append("[ ");
+
+                        for (int i = 0; i < argArray.Length; i++)
+                        {
+                            if (i > 0)
+                                builder.Append(", ");
+
+                            string valueDisplayString = GetDisplayString(argArray.GetValue(i), stringMax);
```

Comment:
As you stated in the discussion. Right now we only use `stringMax` to ensure that strings are not too long. I think we should also apply this to restrict the the length of the result of `builder.ToString` (I've actually not an idea of the default value of `stringMax`, so it could be that it is not sufficient?).

Reply:
I think the default is 40.

Reply:
It looks like the default is actually 0, so it would only truncate if the consumer specifies a limiter when overriding the test name.

Reply:
Actual default is defined by NUnit's default naming template, and is currently 40. Since it's documented to apply only to strings, we either need another setting or should change the docs.

Reply:
Actual default is defined by NUnit's default naming template, and is currently 40. Since it's documented to apply only to strings, we either need another setting or should change the docs.

Reply:
@CharliePoole When I was in there last I thought I noticed that the framework itself does not have a limit by default. Found the PR that removed the limit: https://github.com/nunit/nunit/pull/1556

Reply:
There is no default maximum length any longer. Instead, there is a default naming template, which happens to specify a maximum length of 40 for string arrays. From the user point of view, that's what counts. __Internally__ there is no limit on length.

Reply:
Ah, so the limit is defined by the adapter but not the framework itself.
https://github.com/nunit/nunit3-vs-adapter/pull/189
If any limit comes into play when using the framework with the console or NUnitLite, I'm overlooking it.

Reply:
I think we are getting messed up by two meanings of Default in this context.

1. The default for the framework is "{m}{a}" with no limits. Think of this as the "default default".
2. The default for each runner depends on the runner. If it isn't set, we get the framework default.

The delegation of the actual (effective) default to the runner was decided on about two years ago in the discussions around implementing the template naming feature. The logic was that the framework doesn't care about the default length whereas some runners do.

At one point, I modified the NUnit 3 adapter to use a default length of 40. I think this may now have been overridden and we are using no truncation if the user doesn't specify a template in the `.runsettings` file. This is from a cursory examination, since I don't work on the adapter any longer.

I believe the console now only overrides the "default default" if the user specifies a template on the command-line.

As far as I can see, nobody ever modified the nunitlite runner.

Reply:
Okay, that makes more sense. That means that @yaakov-h is right and the default (defined by the default naming template) is currently 0?

Reply:
Yes, but you have to allow for what the runner sets as the effective default if you are talking to users. My comments about what each runner does right now shouldn't be viewed as definitive, since lots of people have made changes and I'm not following most of them.